### PR TITLE
Parallelize computing predicates in scheduler

### PR DESCRIPTION
**Pull request by wojtek-t**
@davidopp

### CODE REVIEW

File: plugin/pkg/scheduler/generic_scheduler.go [hunk 1]
```diff
@@ -126,51 +125,32 @@ func (g *genericScheduler) selectHost(priorityList schedulerapi.HostPriorityList
 // Filters the nodes to find the ones that fit based on the given predicate functions
 // Each node is passed through the predicate functions to determine if it is a fit
 func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, predicateFuncs map[string]algorithm.FitPredicate, nodes api.NodeList, extenders []algorithm.SchedulerExtender) (api.NodeList, FailedPredicateMap, error) {
+	mu := sync.Mutex{}
```
Comment (gmarek) [inline plugin/pkg/scheduler/generic_scheduler.go]: s/mu/errorsLock/g

Comment (wojtek-t) [inline plugin/pkg/scheduler/generic_scheduler.go]: I'm fine with renaming but errorsLock is not a good name - it is protecting all: filtered, failedPredicateMap and errs

File: plugin/pkg/scheduler/generic_scheduler.go [hunk 2]
```diff
@@ -126,51 +125,32 @@ func (g *genericScheduler) selectHost(priorityList schedulerapi.HostPriorityList
...
-				} else {
-					err := fmt.Errorf("SchedulerPredicates failed due to %v, which is unexpected.", err)
-					return api.NodeList{}, FailedPredicateMap{}, err
-				}
-			}
+	checkNode := func(i int) {
+		nodeName := nodes.Items[i].Name
+		fits, failedPredicate, err := podFitsOnNode(pod, nodeName, nodeNameToInfo[nodeName], predicateFuncs)
+
+		mu.Lock()
```
Comment (gmarek) [inline plugin/pkg/scheduler/generic_scheduler.go]: nit - this probably can be moved inside the if

File: plugin/pkg/scheduler/generic_scheduler.go (hunk 1 above)
Comment (gmarek) [inline plugin/pkg/scheduler/generic_scheduler.go]: Or not... predicateResultsLock?

**Comment by gmarek** [comment]
LGTM

File: plugin/pkg/scheduler/generic_scheduler.go (hunk 1 above)
Comment (wojtek-t) [inline plugin/pkg/scheduler/generic_scheduler.go]: done

**Comment by wojtek-t** [comment]
Comment applied - @gmarek PTAL

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please github issue: #23879

**Comment by davidopp** [comment]
LGTM

Thanks

**Pull request merged**